api/src/main/java/org/apache/iceberg/DataFile.java

```diff
@@ -35,7 +36,7 @@
 /**
  * Interface for data files listed in a table manifest.
  */
-public interface DataFile extends ContentFile<DataFile> {
+public interface DataFile extends ContentFile<DataFile>, Serializable {
```

Comment:
If we are implementing Serializable, should we add serial version uid?

In particular, I have users that run into issues with things that get stored in state having changed version uids when they're not explicitly pinned.